[Discuss] Update Codeowners file

ruflin opened the pull request:
The following is a detailed update of the codeowners file for the different teams. Each top level directory is listed with the team that owns it. In some cases like modules more specific details were added. Every time we see some specific directories belong to a team, the Codeowners file should be updated.

The goal of this PR is to have alignment across all the teams on ownership.

Few additional notes:

* The robots team is not an owner of any files / directories. This is by design as the final ownership of the build is with the data plane team.
* Modules are owned by default by the integrations team. More details must be added here.
* system tests are not covered in details

ruflin commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,102 @@
-/x-pack/packetbeat/ @elastic/security-external-integrations
+### Deploy
+# All changes to the different deployments must be reviewed by data plane team.
+# TODO: This where k8s falls
```
Does the data plan team have to review all changes to k8s for example?

jlind23 replied:
I do not think so.

cmacknz replied:
I don't mind the data plane team being listed as an owner here. SDHs for filebeat/metricbeat on k8s will probably end up with the data plane team, we should know what the deployment set up is.

We could list two codeowners here, one for the beat and one for whoever maintains the deployment types.

ruflin replied:
@cmacknz Do you mean for `/deploy/kubernetes` we add data plane team and the team from @ChrsMark ? I like this idea because it ensures any change to k8s has the eyes from both teams on it.

cmacknz replied:
Yes, let's add both teams.

ruflin commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,102 @@
+/x-pack/filebeat @elastic/elastic-agent-data-plane
+
+# Modules are by defaul owned by the integrations team
+/filebeat/module/ @elastic/integrations
```
@masci Current most modules or owned by integrations or security team is my understanding? Does it make sense that the default ownership is with integrations?

ruflin replied:
Happy to add a list with the detailed owners. @andrewkroh Are you ok with being the default owner?

andrewkroh replied:
I think we should do the effort now of defining the module owners. It will reduce noise on the teams and prevent any ambiguity of responsibility.

ruflin replied:
@masci @andrewkroh Could you put the list together?

andrewkroh replied:
Derived from the elastic/integrations CODEOWNERS and git history.

```
/filebeat/module/apache @elastic/integrations
/filebeat/module/auditd @elastic/security-external-integrations
/filebeat/module/elasticsearch @elastic/integrations
/filebeat/module/haproxy @elastic/integrations
/filebeat/module/icinga @elastic/integrations
/filebeat/module/iis @elastic/integrations
/filebeat/module/kafka @elastic/integrations
/filebeat/module/kibana @elastic/integrations
/filebeat/module/logstash @elastic/integrations
/filebeat/module/mongodb @elastic/integrations
/filebeat/module/mysql @elastic/security-external-integrations
/filebeat/module/nats @elastic/integrations
/filebeat/module/nginx @elastic/integrations
/filebeat/module/osquery @elastic/security-asset-management
/filebeat/module/pensando @elastic/security-external-integrations
/filebeat/module/postgresql @elastic/integrations
/filebeat/module/redis @elastic/integrations
/filebeat/module/santa @elastic/security-external-integrations
/filebeat/module/system @elastic/ecosystem
/filebeat/module/traefik @elastic/integrations
/x-pack/filebeat/module/activemq @elastic/integrations
/x-pack/filebeat/module/aws @elastic/obs-cloud-monitoring
/x-pack/filebeat/module/awsfargate @elastic/obs-cloud-monitoring
/x-pack/filebeat/module/azure @elastic/obs-cloud-monitoring
/x-pack/filebeat/module/barracuda @elastic/security-external-integrations
/x-pack/filebeat/module/bluecoat @elastic/security-external-integrations
/x-pack/filebeat/module/cef @elastic/security-external-integrations
/x-pack/filebeat/module/checkpoint @elastic/security-external-integrations
/x-pack/filebeat/module/cisco @elastic/security-external-integrations
/x-pack/filebeat/module/coredns @elastic/security-external-integrations
/x-pack/filebeat/module/crowdstrike @elastic/security-external-integrations
/x-pack/filebeat/module/cyberarkpas @elastic/security-external-integrations
/x-pack/filebeat/module/cylance @elastic/security-external-integrations
/x-pack/filebeat/module/envoyproxy @elastic/security-external-integrations
/x-pack/filebeat/module/f5 @elastic/security-external-integrations
/x-pack/filebeat/module/fortinet @elastic/security-external-integrations
/x-pack/filebeat/module/gcp @elastic/security-external-integrations
/x-pack/filebeat/module/google_workspace @elastic/security-external-integrations
/x-pack/filebeat/module/ibmmq @elastic/integrations
/x-pack/filebeat/module/imperva @elastic/security-external-integrations
/x-pack/filebeat/module/infoblox @elastic/security-external-integrations
/x-pack/filebeat/module/iptables @elastic/security-external-integrations
/x-pack/filebeat/module/juniper @elastic/security-external-integrations
/x-pack/filebeat/module/microsoft @elastic/security-external-integrations
/x-pack/filebeat/module/misp @elastic/security-external-integrations
/x-pack/filebeat/module/mssql @elastic/integrations
/x-pack/filebeat/module/mysqlenterprise @elastic/security-external-integrations
/x-pack/filebeat/module/netflow @elastic/security-external-integrations
/x-pack/filebeat/module/netscout @elastic/security-external-integrations
/x-pack/filebeat/module/o365 @elastic/security-external-integrations
/x-pack/filebeat/module/okta @elastic/security-external-integrations
/x-pack/filebeat/module/oracle @elastic/security-external-integrations
/x-pack/filebeat/module/panw @elastic/security-external-integrations
/x-pack/filebeat/module/proofpoint @elastic/security-external-integrations
/x-pack/filebeat/module/rabbitmq @elastic/integrations
/x-pack/filebeat/module/radware @elastic/security-external-integrations
/x-pack/filebeat/module/snort @elastic/security-external-integrations
/x-pack/filebeat/module/snyk @elastic/security-external-integrations
/x-pack/filebeat/module/sonicwall @elastic/security-external-integrations
/x-pack/filebeat/module/sophos @elastic/security-external-integrations
/x-pack/filebeat/module/squid @elastic/security-external-integrations
/x-pack/filebeat/module/suricata @elastic/security-external-integrations
/x-pack/filebeat/module/threatintel @elastic/security-external-integrations
/x-pack/filebeat/module/tomcat @elastic/security-external-integrations
/x-pack/filebeat/module/zeek @elastic/security-external-integrations
/x-pack/filebeat/module/zookeeper @elastic/integrations
/x-pack/filebeat/module/zoom @elastic/security-external-integrations
/x-pack/filebeat/module/zscaler @elastic/security-external-integrations
```

ruflin replied:
Thanks you @andrewkroh ❤️

ruflin commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,102 @@
+/x-pack/metricbeat/ @elastic/elastic-agent-data-plane
+
+# Modules are owned by the integrations team by default
+/metricbeat/module/ @elastic/integrations
```
@masci Same as above.

cmacknz replied:
The system module should be the data plane team https://github.com/elastic/beats/tree/main/metricbeat/module/system

ruflin replied:
Added the system module to the data plane team.

ruflin commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,102 @@
-/heartbeat/ @elastic/uptime
+#### monitors.d
+# TODO: Does this belong to synthetics? Why is this in the top dir?
+/monitors.d/ @elastic/uptime
```
@andrewvc This kind of looks off on this level.

andrewvc replied:
That's not right, we can remove this. That directory only exists in `heartbeat` and `x-pack/heartbeat` which should already be covered.

ruflin replied:
Can you follow up with a PR removing the directory?

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30793/detail/PR-30793/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30793/detail/PR-30793/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30793/detail/PR-30793/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-30793/detail/PR-30793/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_30793.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2022-03-14T08:08:08.496Z&rangeTo=2022-03-14T08:28:08.496Z&transactionName=BUILD+Beats/beats/PR-{number}&transactionType=job&latencyAggregationType=avg&traceId=eeacaec0e695a6071911ed0da17aa8f5&transactionId=ef4c95178f1c8c27)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-03-14T08:18:08.496+0000


* Duration: 14 min 3 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 3  |
| Skipped      | 0 |
| Total        | 3   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->












## :green_heart: Flaky test report
Tests succeeded.



## :robot: GitHub comments

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.


- `/package` : Generate the packages and run the E2E tests.


- `/beats-tester` : Run the installation tests with beats-tester.


- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)




<!--COMMENT_GENERATED_WITH_ID_comment.id-->

ph reviewed (commented):
We might want to add `libbeat/management` to the control plane. This PR will go in 7.x too right?

ruflin commented:
Good question from @ph if this should be backported? So far I was thinking no as the changes to 7.x are hopefully minimal. I would go so far that all changes going into 7.x must be reviewed by the control and data plane team from now on.

I'll add the libbeat/management change.

jlind23 commented:
I would vote for a backport as it won't have a huge impact and may avoid potential "silent" merge 👍🏼

cmacknz commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,103 @@
+/filebeat @elastic/elastic-agent-data-plane
+/x-pack/filebeat @elastic/elastic-agent-data-plane
+
+# Modules are by defaul owned by the integrations team
```
```suggestion
# Modules are by default owned by the integrations team
```

cmacknz commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,103 @@
+/x-pack/heartbeat/ @elastic/uptime
+
+### Journalbeat
+# Journalbeat is owned by the data plan team
```
```suggestion
# Journalbeat is owned by the data plane team
```

andrewvc commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,102 @@
+
+
+### Heartbeat
+# TODO: Is this team still accurate?
```
yep!

ruflin commented:
Thanks for the list @andrewkroh , updated the change with it.

andrewkroh commented:
If this is backported will it affect the ability to merge backport PRs without review? Today, we don't require any review of backports.

ruflin commented:
@v1v Will this have affects on the backporting tool?

v1v commented:
> @v1v Will this have affects on the backporting tool?

I believe so, it's a native support from GitHub, see https://github.com/Mergifyio/mergify-engine/issues/464

> If this is backported will it affect the ability to merge backport PRs without review? Today, we don't require any review of backports.

IIUC, that's not the case with the existing branch protection:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/2871786/159271818-0c2f59a9-c8bc-47a7-9b07-d40d8b34c8d0.png">

That's the one for the [`8.*`](https://github.com/elastic/beats/settings/branch_protection_rules/22874483) branches

ruflin commented:
Based on the comment from @v1v it seems, we don't require reviews from Code owners on "backport" branches. This is great for getting this change in but TBH it makes a bit nervous as it means in release branches changes could be merged without a code owner review. As elasticmachine can merge without reviews, I guess we could still turn this on and the automatic backporting would keep working?

ruflin commented:
I updated the list with the details provided by @andrewkroh . I also added `@elastic/security-asset-management` permissions to the beats repo as this is required for the CODEOWNERS file to work. Single users like @aleksmaus already had write access to the repo but not the team so this wasn't an issue previously.

I know I have several 👍  on the PR but I made again a big change with the list from Andrew. Would be good to get someone to double check again.

@andrewkroh I have the suspicion you used some script for the codeowners list, can you share it if you have one? In any case, we should find a way to automate this. I would be nice if we could have local CODEOWNERS file so it can be maintained locally without having to update the global one manually.

andrewkroh commented:
> I have the suspicion you used some script for the codeowners list, can you share it if you have one?

There was a big bash command, but it's gone now 😞 . Basically it listed the filebeat modules, grepped each of their names in the integrations repo's codeowners file, take the first result, and put it into that list. Then I manually reviewed the list and checked the history of modules that were not found.

andrewkroh commented on .github/CODEOWNERS:
```diff
@@ -1,55 +1,203 @@
 
-# * @elastic/beats
+# The beats repository is owned by the @elastic/elastic-agent-data-plane Many teams contribute to it. The goal
+# is to cover all directories in the CODEOWNERS file. The list is sorted alphabetically by directory and sub directories.
```
> The list is sorted alphabetically by directory and sub directories.

That's not true given the x-pack dirs that are listed near the beginning. Maybe we should drop most of the comments and do a `sort` on the contents.

Comments like `# All changes to the different deployments must be reviewed by data plane team.` do not add much value IMO, you can see that fact codified in `/deploy/ @elastic/elastic-agent-data-plane`.

ruflin replied:
I unfortunately agree. When I started there were like 20 lines and it was easy to see the groups by team but then things exploded.

What I will do is merge the current PR as is but I want to follow up with automation which automatically will mean, sorting is in place and all comments will disappear.
